utils.go

```diff
@@ -40,6 +41,26 @@ func GetMethod(instance interface{}, methodName string) interface{} {
 	return nil
 }
 
+// GetPrivateMemberMethod resolve a method from an instance, include private method.
+//
+// F must fit the shape of specific method, include receiver as the first argument.
+// Especially, the receiver can be replaced as interface when F is declaring,
+// this will be very useful when receiver type is not exported for other packages.
+//
+// for example:
+//
+//	GetPrivateMemberMethod[func(*bytes.Buffer) bool](&bytes.Buffer{}, "empty")
+//	GetPrivateMemberMethod[func(hash.Hash, []byte) [sha256.Size]byte](sha256.New(), "checkSum")
+func GetPrivateMemberMethod[F interface{}](instance interface{}, methodName string) interface{} {
```

Comment:
generic is not supported in go 1.17 and below and could a compile problem

Reply:
I notice go.mod require 1.18 version :), this method is not requried, I'll move the method to a new file add the build version hint .